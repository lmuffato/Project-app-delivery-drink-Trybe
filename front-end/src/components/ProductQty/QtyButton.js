import React from 'react';
import PropTypes from 'prop-types';
import { TiMinus as Minus, TiPlus as Plus } from 'react-icons/ti';

function QtyButton({ position, onClick, id }) {
  switch (position) {
  case 'remove':
    return (
      <button
        className="remove"
        type="button"
        onClick={ onClick }
        data-testid={ `customer_products__button-card-rm-item-${id}` }
      >
        <Minus />
      </button>
    );
  case 'add':
    return (
      <button
        className="add"
        type="button"
        onClick={ onClick }
        data-testid={ `customer_products__button-card-add-item-${id}` }
      >
        <Plus />
      </button>
    );
  default:
    return null;
  }
}

QtyButton.propTypes = {
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  position: PropTypes.oneOf(['remove', 'add']).isRequired,
  onClick: PropTypes.func,
};

QtyButton.defaultProps = {
  onClick: () => { },
};

export default QtyButton;
