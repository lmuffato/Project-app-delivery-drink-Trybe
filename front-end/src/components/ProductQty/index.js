import React from 'react';
import PropTypes from 'prop-types';
import { ProductQtyContainer } from '../../styles/baseComponents';
import QtyButton from './QtyButton';

function ProductQty({ label, onMinus, onPlus, value, id }) {
  return (
    <ProductQtyContainer>
      <p
        className="label"
        data-testid={ `customer_products__element-card-title-${id}` }
      >
        {label}
      </p>
      <div className="bts">
        <QtyButton
          id={ id }
          position="remove"
          onClick={ onMinus }
        />
        <div
          className="center"
          data-testid={ `customer_products__input-card-quantity-${id}` }
        >
          {value}
        </div>
        <QtyButton
          id={ id }
          position="add"
          onClick={ onPlus }
        />
      </div>
    </ProductQtyContainer>
  );
}

ProductQty.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  onMinus: PropTypes.func.isRequired,
  onPlus: PropTypes.func.isRequired,
};

export default ProductQty;
