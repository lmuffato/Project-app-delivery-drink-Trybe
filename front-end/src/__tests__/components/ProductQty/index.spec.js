import React from 'react';
import { render } from '@testing-library/react';
import ProductQty from '../../../components/ProductQty/index';

describe('Snapshot test', () => {
  it('renders correctly', () => {
    const JSON = render(
      <ProductQty
        id="1"
        label="Label Teste"
        value="Value Teste"
        onAdd={ jest.fn() }
        onRemove={ jest.fn() }
      />,
    );
    expect(JSON).toMatchSnapshot();
  });
});
