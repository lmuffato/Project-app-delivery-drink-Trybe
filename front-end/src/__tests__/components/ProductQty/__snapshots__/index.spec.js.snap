// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKwDye ZwEEK"
      >
        <p
          class="label"
          data-testid="customer_products__element-card-title-1"
        >
          Label Teste
        </p>
        <div
          class="bts"
        >
          <button
            class="remove"
            data-testid="customer_products__button-card-rm-item-1"
            type="button"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 11h-12c-1.104 0-2 .896-2 2s.896 2 2 2h12c1.104 0 2-.896 2-2s-.896-2-2-2z"
              />
            </svg>
          </button>
          <div
            class="center"
            data-testid="customer_products__input-card-quantity-1"
          >
            Value Teste
          </div>
          <button
            class="add"
            data-testid="customer_products__button-card-add-item-1"
            type="button"
          >
            <svg
              baseProfile="tiny"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1.2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 10h-4v-4c0-1.104-.896-2-2-2s-2 .896-2 2l.071 4h-4.071c-1.104 0-2 .896-2 2s.896 2 2 2l4.071-.071-.071 4.071c0 1.104.896 2 2 2s2-.896 2-2v-4.071l4 .071c1.104 0 2-.896 2-2s-.896-2-2-2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKwDye ZwEEK"
    >
      <p
        class="label"
        data-testid="customer_products__element-card-title-1"
      >
        Label Teste
      </p>
      <div
        class="bts"
      >
        <button
          class="remove"
          data-testid="customer_products__button-card-rm-item-1"
          type="button"
        >
          <svg
            baseProfile="tiny"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 11h-12c-1.104 0-2 .896-2 2s.896 2 2 2h12c1.104 0 2-.896 2-2s-.896-2-2-2z"
            />
          </svg>
        </button>
        <div
          class="center"
          data-testid="customer_products__input-card-quantity-1"
        >
          Value Teste
        </div>
        <button
          class="add"
          data-testid="customer_products__button-card-add-item-1"
          type="button"
        >
          <svg
            baseProfile="tiny"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 10h-4v-4c0-1.104-.896-2-2-2s-2 .896-2 2l.071 4h-4.071c-1.104 0-2 .896-2 2s.896 2 2 2l4.071-.071-.071 4.071c0 1.104.896 2 2 2s2-.896 2-2v-4.071l4 .071c1.104 0 2-.896 2-2s-.896-2-2-2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
